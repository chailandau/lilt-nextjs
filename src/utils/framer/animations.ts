export const menuAnimations = {
    closed: {
        y: '-100%',
        opacity: 0,
        height: 0,
        transition: {
            delay: 0.15,
            length: 1
        }
    },
    open: {
        y: 0,
        opacity: 1,
        height: '100vh',
        transition: {
            type: 'spring',
            duration: 0.4
        }
    }
};

export const dropdownAnimations = {
    closed: {
        y: '15%',
        opacity: 0,
        transition: {
            duration: 0.25
        }
    },
    open: {
        y: 0,
        opacity: 1,
        transition: {
            type: 'spring',
            duration: 0.25
        }
    }
};

export const iconFlip = {
    closed: { rotate: 0, transition: { type: 'spring', duration: 0.25 } },
    open: {
        rotate: 180,
        transition: {
            type: 'spring',
            duration: 0.25
        }
    }
};
