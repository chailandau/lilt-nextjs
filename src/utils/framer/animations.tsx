export const menuAnimations = {
    closed: {
        y: '-100%',
        opacity: 0,
        height: 0,
        transition: {
            delay: 0.15,
            length: 1
        }
    },
    open: {
        y: 0,
        opacity: 1,
        height: '100vh',
        transition: {
            type: 'spring',
            duration: 0.4
        }
    }
};

export const dropdownAnimations = (isLaptop: boolean | null) => ({
    closed: {
        y: '15%',
        x: isLaptop ? 'calc(-50% - 6px)' : 0,
        opacity: 0,
        transition: {
            duration: 0.25
        }
    },
    open: {
        y: 0,
        x: isLaptop ? 'calc(-50% - 6px)' : 0,
        opacity: 1,
        transition: {
            type: 'spring',
            duration: 0.25
        }
    }
});

export const iconFlip = {
    closed: { rotate: 0, transition: { type: 'spring', duration: 0.25 } },
    open: {
        rotate: 180,
        transition: {
            type: 'spring',
            duration: 0.25
        }
    }
};
