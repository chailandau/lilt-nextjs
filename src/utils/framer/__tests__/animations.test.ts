import * as framerAnimations from '../animations';
import { dropdownAnimations } from '../animations';

describe('dropdownAnimations', () => {
    it('returns correct animation when laptop is true', () => {
        const isLaptop = true;
        const result = dropdownAnimations(isLaptop);
        expect(result).toMatchSnapshot();
    });
    it('returns correct animation when laptop is false', () => {
        const isLaptop = false;
        const result = dropdownAnimations(isLaptop);
        expect(result).toMatchSnapshot();
    });
});
describe('Framer animations', () => {
    Object.keys(framerAnimations).forEach((animationName) => {
        const animation =
            framerAnimations[animationName as keyof typeof framerAnimations];
        it(`${animationName} matches snapshot`, () => {
            expect(animation).toMatchSnapshot();
        });
    });
});
