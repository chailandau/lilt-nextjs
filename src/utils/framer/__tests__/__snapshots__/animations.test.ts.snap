// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Framer animations dropdownAnimations matches snapshot 1`] = `[Function]`;

exports[`Framer animations iconFlip matches snapshot 1`] = `
{
  "closed": {
    "rotate": 0,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
  },
  "open": {
    "rotate": 180,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
  },
}
`;

exports[`Framer animations menuAnimations matches snapshot 1`] = `
{
  "closed": {
    "height": 0,
    "opacity": 0,
    "transition": {
      "delay": 0.15,
      "length": 1,
    },
    "y": "-100%",
  },
  "open": {
    "height": "100vh",
    "opacity": 1,
    "transition": {
      "duration": 0.4,
      "type": "spring",
    },
    "y": 0,
  },
}
`;

exports[`dropdownAnimations returns correct animation when laptop is false 1`] = `
{
  "closed": {
    "opacity": 0,
    "transition": {
      "duration": 0.25,
    },
    "x": 0,
    "y": "15%",
  },
  "open": {
    "opacity": 1,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
    "x": 0,
    "y": 0,
  },
}
`;

exports[`dropdownAnimations returns correct animation when laptop is true 1`] = `
{
  "closed": {
    "opacity": 0,
    "transition": {
      "duration": 0.25,
    },
    "x": "calc(-50% - 6px)",
    "y": "15%",
  },
  "open": {
    "opacity": 1,
    "transition": {
      "duration": 0.25,
      "type": "spring",
    },
    "x": "calc(-50% - 6px)",
    "y": 0,
  },
}
`;
